src/Stryker.Core/Stryker.Core/Initialisation/ProjectFileReader.cs

```diff
@@ -104,24 +104,29 @@ private IAnalyzerResult SelectAnalyzerResult(IAnalyzerResults analyzerResults, s
         {
             if (targetFramework == null)
             {
-                return analyzerResults.First(e => e.TargetFramework != null);
+                var analyzerResult = analyzerResults.FirstOrDefault(e => e.TargetFramework is not null);
+
+                if (analyzerResult is null)
+                {
+                    throw new InputException("No analyzer result with a valid target framework could be found.");
+                }
+
+                return analyzerResult;
             }
 
             var analyzerResultForFramework = analyzerResults.SingleOrDefault(result => result.TargetFramework == targetFramework);
-            if (analyzerResultForFramework != null)
+            if (analyzerResultForFramework is not null)
             {
                 return analyzerResultForFramework;
             }
-            else
-            {
-                var firstAnalyzerResult = analyzerResults.First();
-                _logger.LogWarning(
-                    "The configured target framework '{0}' isn't available for this project. " +
-                    "It will be built against the first framework available " +
-                    "which is {1}.", targetFramework, firstAnalyzerResult.TargetFramework);
 
-                return firstAnalyzerResult;
-            }
+            var firstAnalyzerResult = analyzerResults.First(e => e.TargetFramework is not null);
```

Comment:
We can extract https://github.com/stryker-mutator/stryker-net/pull/2030/files#diff-b73ef193446dc576c0b7b41967963436819ee09852c44e4c10d4e1df4e4d726fR107-R114 into a function and reuse it here, since we should once again test if we can actually find a valid analyzer result and if not we should throw.

Reply:
If we invert the if (targetFramework == null) perhaps we can avoid this duplicate check completely.

Reply:
yes, logic could be:
1. if a specific `targetframework` is expected
    1. try to find a matching result
    2. if none find, log a warning: `The configured target...
1. find the first result with a valid framework
    1. if not, raise an exception

Reply:
I started implementing that logic and instead found a way to simplify. Please let me know.

Reply:
Looks good to me